### PR TITLE
[barfiller] Fix EXP rate calculation

### PR DESCRIPTION
**Pull request by nmarigoni**
Fixed barfiller's EXP rate calculation to recognize gains from EXP chain messages, and to use current time rather than CPU time for calculations.

### CODE REVIEW

**Comment by RubenatorX** [comment]
edit: actually I'm mistaken (was corrected by another dev) -- they are the same units, but clock has milliseconds so nvm on all that. You can change it to os.time if you like, seems sufficient for this use case
~~1) clock and time have completely different units, you can't just swap them out and expect them to work as intended when for example compared to `600` (e.g. line `254`)
2) on windows, the os clock function actually just returns the time (rather than clock cycles) anyway.
3) os.time is in sections, and therefore not as granular as os.clock, so if you want to change it to anything, you'd probably use `sockets.gettime()` which has millisecond precision (although per 1 & 2, shouldn't need to change it at all necesarilly)~~

File: addons/barfiller/statics.lua [hunk 1]
```diff
@@ -234,8 +234,8 @@ function display_help()
 end
 
 function exp_msg(val,msg)
-    local t = os.clock()
-    if msg == 8 or msg == 105 then
+    local t = os.time()
+    if msg == 8 or msg == 105 or msg ==253 then
```
Comment (lili-ffxi) [inline addons/barfiller/statics.lua]: Missing space between `==` and the value

**Comment by RubenatorX** [comment]
I'm mistaken, adjusted comment above to reflect correction
Although point number 2 is correct.
We are unsure if there is a difference in the `os.clock` implementation in Wine (for the Linux users out there), but os.time should be "more correct" for Linux at least

**Comment by nmarigoni** [comment]
> I'm mistaken, adjusted comment above to reflect correction Although point number 2 is correct. We are unsure if there is a difference in the `os.clock` implementation in Wine (for the Linux users out there), but os.time should be "more correct" for Linux at least

Thanks for the feedback. The change to os.time was intended to be a housekeeping rather than functional change, as the Lua documentation did suggest is was the more appropriate function. I was not aware of the difference in functionality between platforms and am happy to defer if there are any concerns. However, it sounds like the view at this time is that using os.time is appropriate, so I'll leave as-is absent further feedback on this point.

File: addons/barfiller/statics.lua (hunk 1 above)
Comment (nmarigoni) [reply addons/barfiller/statics.lua]: Thanks! Has been corrected.

**Pull request merged**